### PR TITLE
bugfix and test case for run_examples

### PR DESCRIPTION
**Pull request by amcdavid**
Don't pass `show` to `pkgload::run_example`.  Also adds a regression test for run_examples.

### CODE REVIEW

**Comment by jimhester** [comment]
This looks OK, but your test needs to be tweaked, you just need to escape the special characters in the regex (https://travis-ci.org/hadley/devtools/jobs/195934346#L895-L896)

**Comment by amcdavid** [comment]
Hrm, weird.  I added `fixed=TRUE` and it worked on my machine.  Will try escaping and removing `fixed`.

**Comment by amcdavid** [comment]
I am buffaloed as to why this fails on the CI.  R CMD check on the built package works fine on my system.

**Comment by jimhester** [comment]
You can safely ignore the Appveyor failures they are unrelated to this change.

File: tests/testthat/test-examples.R [hunk 1]
```diff
@@ -0,0 +1,6 @@
+context('Examples')
+
+test_that('Can run an example', {
+    pkg <- "testHelp"
+    expect_output(run_examples(pkg=pkg), "You called foofoo.", fixed=TRUE)
```
Comment (hadley) [inline tests/testthat/test-examples.R]: Can you please put spaces around `=` and dedent `}`

File: NEWS.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 Workaround a regression in Rcpp::compileAttributes.
 Add trimws implementation for R 3.1 support.
 
+* `run_examples` was mistakenly passing `show` to
```
Comment (hadley) [inline NEWS.md]: Can you please move this to the correct place once you merge/rebase?

File: tests/testthat/testHelp/R/foofoo.r [hunk 3]
```diff
@@ -3,5 +3,7 @@
 #' The purpose of this function is to test out \code{help} and \code{?} from
 #' devtools.
 #'
+#' @examples
+#' stopifnot(foofoo()=='You called foofoo.')
```
Comment (hadley) [inline tests/testthat/testHelp/R/foofoo.r]: Please add spaces around `==`

**Comment by hadley** [comment]
Fixes #1449

**Comment by hadley** [comment]
@amcdavid are you interested in finishing this off?

**Comment by amcdavid** [comment]
Wrangling this pull RQ exceeds the bounds of my github-fu.  I'll leave this to someone more experienced to finish off.

**Comment by hadley** [comment]
No problems - @jimhester can finish it off. Sorry for letting it languish for so long!

**Comment by Neil-Schneider** [comment]
I attempted resolving this in #1580, but ran into some odd behavior from the warning if `show` was not missing.

**Comment by jimhester** [comment]
I merged this manually at https://github.com/hadley/devtools/commit/0128ed8ef52e81fcea270bf1b8e2ba40cf95db2c

**Comment by jimhester** [comment]
Thanks for the PR and for your patience!

**Pull request closed without merging**